packages/templates/blog/template/src/bootstrap.js

```diff
@@ -72,119 +71,172 @@ function getFileData(fileName) {
   };
 }
 
+async function uploadFile(file, name) {
+  return strapi
+    .plugin("upload")
+    .service("upload")
+    .upload({
+      files: file,
+      data: {
+        fileInfo: {
+          alternativeText: name,
+          caption: name,
+          name,
+        },
+      },
+    });
+}
+
 // Create an entry and attach files if there are any
-async function createEntry({ model, entry, files }) {
+async function createEntry({ model, entry }) {
   try {
-    if (files) {
-      for (const [key, file] of Object.entries(files)) {
-        // Get file name without the extension
-        const [fileName] = file.name.split('.');
-        // Upload each individual file
-        const uploadedFile = await strapi
-          .plugin("upload")
-          .service("upload")
-          .upload({
-            files: file,
-            data: {
-              fileInfo: {
-                alternativeText: fileName,
-                caption: fileName,
-                name: fileName,
-              },
-            },
-          });
-
-        // Attach each file to its entry
-        set(entry, key, uploadedFile[0].id);
-      }
-    }
-
     // Actually create the entry in Strapi
-    const createdEntry = await strapi.entityService.create(
-      `api::${model}.${model}`,
-      {
-        data: entry,
-      }
-    );
-  } catch (e) {
-    console.log("model", entry, e);
+    await strapi.entityService.create(`api::${model}.${model}`, {
+      data: entry,
+    });
+  } catch (error) {
+    console.error({ model, entry, error });
   }
 }
 
 async function importCategories() {
-  return Promise.all(
-    categories.map((category) => {
-      return createEntry({ model: "category", entry: category });
-    })
-  );
+  for (const category of categories) {
```

Comment:
I'm curious why you made this change. Was there an issue with the parallelization? (same question for all content types)

Reply:
You know it's a good question..I switched the `.map` to a `for...of` loop for the articles because I prefer it when working with `async/await`. That's when I also removed the `Promise.all` and I think I eventually I changed it everywhere for consistency, but really I had no real reason to remove it. 

I was curious though and I did a quick test on the `articles` creation with some interesting and maybe questionable results 🤔 :

test articles 1 (for of sequentially): 1.247s
test articles 2 (map + promise.all): 2.010s
test articles 3 (for of + promise.all): 1.351s

It should be quicker to process everything in parallel but here that doesn't seem to be the case. I'm just wondering if we are really concerned about performance here. The data is static and the arrays are never very big. I don't really have a preference but I'm not convinced the parallelization is actually adding a lot of value.

Reply:
Sounds good to me then, I agree it's more readable like this